File: 1800-maximum-ascending-subarray-sum/main.go
func maxAscendingSum(nums []int) int {
    _max := 0
    currSubSum := nums[0]

    for i := 1; i < len(nums); i++ {
        if nums[i] <= nums[i - 1] {
            _max = max(_max, currSubSum)
            currSubSum = 0
        }

        currSubSum += nums[i]
    }

    return max(_max, currSubSum)
}

func max(a, b int) int {
    if a < b {
        return b
    }
    return a
}

File: 1800-maximum-ascending-subarray-sum/main.py
class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        _max = 0
        curr_sub_sum = nums[0]

        for curr_idx in range(1, len(nums)):

            if nums[curr_idx] <= nums[curr_idx - 1]:
                _max = max(_max, curr_sub_sum)
                curr_sub_sum = 0 

            curr_sub_sum += nums[curr_idx]

        return max(_max, curr_sub_sum)
